pkg/image/validation/kubernetes.go

```diff
@@ -44,21 +46,49 @@ func isKubernetesDefined(k8s *image.Kubernetes) bool {
 	return k8s.Version != ""
 }
 
-func validateNodes(k8s *image.Kubernetes) []FailedValidation {
+func validateNetwork(k8s *image.Kubernetes) []FailedValidation {
 	var failures []FailedValidation
+	var vipAddress = k8s.Network.APIVIP
+
+	if vipAddress == "" {
+		numNodes := len(k8s.Nodes)
+		// Single node cluster, apiVIP is optional, check only multi-node
+		if numNodes > 1 {
+			failures = append(failures, FailedValidation{
+				UserMessage: "The 'apiVIP' field is required in the 'network' section when defining entries under 'nodes'.",
+			})
+		}
 
-	numNodes := len(k8s.Nodes)
-	if numNodes <= 1 {
-		// Single node cluster, node configurations are not required
 		return failures
 	}
 
-	if k8s.Network.APIVIP == "" {
+	ip, err := netip.ParseAddr(vipAddress)
+	if err != nil {
+		msg := fmt.Sprintf("Invalid cluster API address ('%s')", strings.ToLower(vipAddress))
 		failures = append(failures, FailedValidation{
-			UserMessage: "The 'apiVIP' field is required in the 'network' section when defining entries under 'nodes'.",
+			UserMessage: msg,
```

Comment:
We must return after this otherwise we'll run into nil pointer dereference below.

Reply:
Good catch, it got lost during the copy to validation!